### PR TITLE
MM-52: Added testcase for admin restriction configuration

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
### Description
Added testcase for admin restriction configuration

### CODE REVIEW

File: server/command_hooks_test.go [hunk 1]
```diff
@@ -424,3 +425,98 @@ func TestExecuteCommand(t *testing.T) {
...
+			name:                    "Admin restrictions enabled, has permission to system",
+			enableAdminRestrictions: true,
+			userID:                  "user2",
+			channelID:               "channel2",
+			hasPermissionToChannel:  false,
+			hasPermissionToSystem:   true,
+			expected:                true,
+		},
+		{
+			name:                    "Admin restrictions enabled, lacks both permissions",
```
Comment (raghavaggarwal2308) [inline server/command_hooks_test.go]: Elaborate the msg

File: server/command_hooks_test.go [hunk 2]
```diff
@@ -424,3 +425,98 @@ func TestExecuteCommand(t *testing.T) {
...
+		userID                  string
+		channelID               string
+		hasPermissionToChannel  bool
+		hasPermissionToSystem   bool
+		expected                bool
+	}{
+		{
+			name:                    "Admin restrictions enabled, has permission to channel",
+			enableAdminRestrictions: true,
+			userID:                  "user1",
```
Comment (raghavaggarwal2308) [inline server/command_hooks_test.go]:
```suggestion
			userID:                  "mockUserID1",
```

File: server/command_hooks_test.go [hunk 3]
```diff
@@ -424,3 +425,98 @@ func TestExecuteCommand(t *testing.T) {
...
+			channelID:               "channel1",
```
Comment (raghavaggarwal2308) [inline server/command_hooks_test.go]:
```suggestion
			channelID:               "mockChanneID1",
```

File: server/command_hooks_test.go [hunk 4]
```diff
@@ -424,3 +425,98 @@ func TestExecuteCommand(t *testing.T) {
...
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.enableAdminRestrictions {
+				mockUser.On("HasPermissionToChannel", tt.userID, tt.channelID, model.PermissionManageChannelRoles).Return(tt.hasPermissionToChannel)
+				if !tt.hasPermissionToChannel {
+					mockUser.On("HasPermissionTo", tt.userID, model.PermissionManageSystem).Return(tt.hasPermissionToSystem)
+				}
```
Comment (raghavaggarwal2308) [inline server/command_hooks_test.go:510]: Does it need to be nested?

File: server/command_hooks_test.go [hunk 5]
```diff
@@ -424,3 +425,98 @@ func TestExecuteCommand(t *testing.T) {
...
+			hasPermissionToChannel:  false,
+			hasPermissionToSystem:   false,
+			expected:                true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.enableAdminRestrictions {
+				mockUser.On("HasPermissionToChannel", tt.userID, tt.channelID, model.PermissionManageChannelRoles).Return(tt.hasPermissionToChannel)
```
Comment (raghavaggarwal2308) [inline server/command_hooks_test.go:507]: I think directly mocking the API methods will also work. As p.client is using that internally. Please check

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: server/command_hooks_test.go (hunk 4 above)
Comment (Kshitij-Katiyar) [reply server/command_hooks_test.go:510]: Yes, if `hasPermissionToChannel` is true then only the call to `HasPermissionTo` is made.

File: server/command_hooks_test.go (hunk 5 above)
Comment (Kshitij-Katiyar) [reply server/command_hooks_test.go:507]: Will need to discuss

Comment (raghavaggarwal2308) [reply server/command_hooks_test.go:507]: discussed

**Review by raghavaggarwal2308 — APPROVED** [review]

**Comment by Kshitij-Katiyar** [comment]
Created a PR with all these changes on Mattermost Remote hence closing this one

**Pull request closed without merging**